### DISCUSSION TITLE
AltStore Installation Error: "File doesn't exist" (RESOLVED)

### DISCUSSION THREAD

**Discussion by geekalert** [utmapp/UTM#3572] [UTM Help (iOS)]
**Describe the issue**
Error "The file doesn't exist" appears when attempting to install UTM SE from AltStore source, https://alt.getutm.app.

**Configuration (required)**
* UTM Version: N/A - cannot install
* OS Version: iPadOS 15.3
* Device Model: iPad Pro 2018 12.9"
* Is it jailbroken (name jailbreak used)? No
* How did you install UTM? Always used AltStore in the past

**Crash log**
If the app crashed, you need a crash log. To get your crash log, open the Settings app and browse to `Privacy -> Analytics & Improvements -> Analytics Data` and find the latest entry for UTM. You should export the text and attach it here.

-> N/A; I do not have a crash log because I cannot install the app.

**Debug log**
For all issues, _including_ crashes, you should attach a debug log. Open UTM, and open the settings for the VM you wish to launch. Near the bottom of the configuration options (or at the top of the `QEMU` page) is `Debug Log`. Turn it on and save the VM. After you experience the issue, quit the VM and re-launch UTM. Open the VM settings again and select `Export Log...` and attach it here.

-> N/A; same as above.

**Upload VM**
(Optional) If possible, upload the `config.plist` inside the `.utm`. If you do not have this, you can upload the entire `.utm` but note this contains your personal data. Since GitHub has an attachment size limit, you may want to upload to another service such as Google Drive. Link it here.

-> N/A

**Comment by osy** [comment]
Make sure you’re not out of disk space.

**Reply by geekalert** [reply]
Negative. I am using 40.5 GB of 512 GB. It was literally installed 10 minutes earlier, but AltStore had expired so I decided to uninstall and reinstall AltStore and my apps — as I have done *many* times over the past couple of years. I've been a user since UTM was super early days.

**Here are the full details:**

- I can add the repo, no problem and can see both UTM and UTM SE packages as "installable" in AltStore, as expected.
- I can install UTM (non-SE version) through AltStore without issue (but I cannot use it because I am on v15.3).
- I can install UTM SE by downloading the IPA and sideloading it through AltStore, without error.
- When I try to install SE from AltStore, I get the error that the file is not found, as stated.
- I am a *long-time* regular user of both AltStore and UTM, experienced with installing and re-installing both. 
- I first noted this issue a few weeks ago when I was trying to apply an "update" AltStore said it had. Since I already had it installed and it was functioning correctly, I didn't worry about it at the time. I thought someone else would have been reported by now and the connection to the installer would have been fixed.

This is a legit issue, not really a "discussion". Something is definitively wrong with how the UTM AltStore source is trying to distribute the file for the SE version ONLY. That's suspect. I would appreciate if you looked into it @osy . I have all of my mobile dev environments in UTM because it's an absolute *brilliant* utility. You have made a truly amazing, invaluable tool. It's would be a serious bummer to be without it for long. In the meantime, I will install from IPA, but please see if you can correct the AltStore connection to file.

Thank you.

**Reply by osy** [reply]
It's a discussion because it's not a UTM issue. As in there's no code changes to UTM. Don't take it personally.

I think the repo link is outdated. I triggered a refresh in the repo generation. Try again and let me know.

**Reply by geekalert** [reply]
Not personal, just didn't want it to get sidelined as a discussion vs something that needed attention. I appreciate you looking into it, and I can confirm that the issue has been resolved. Thank you, sir. Keep up the amazing work!
